app/src/org/odk/collect/android/utilities/FileUtils.java

```diff
@@ -217,21 +217,24 @@ public static Bitmap getBitmapScaledToDisplay(File f, int screenHeight, int scre
             scale = 1;
         }
 
-        return performSafeScaleDown(f, scale);
+        return performSafeScaleDown(f, scale, 0);
     }
 
     /**
      * Returns a scaled-down bitmap for the given image file, progressively increasing the
      * scale-down factor by 1 until allocating memory for the bitmap does not cause an OOM error
      */
-    private static Bitmap performSafeScaleDown(File f, int scale) {
+    private static Bitmap performSafeScaleDown(File f, int scale, int depth) {
+        if (depth == 8) {
+            // Limit the number of recursive calls
+            return null;
+        }
         BitmapFactory.Options options = new BitmapFactory.Options();
         options.inSampleSize = scale;
         try {
             return BitmapFactory.decodeFile(f.getAbsolutePath(), options);
         } catch (OutOfMemoryError e) {
-            scale++;
-            return performSafeScaleDown(f, scale);
+            return performSafeScaleDown(f, ++scale, ++depth);
```

Comment:
Not sure what you mean, what about this case in particular makes that undesirable? I switched to using that because now that both variables needed to be incremented, it's much more succinct.